RiotSwiftUI/Modules/Room/UserSuggestion/Service/UserSuggestionServiceProtocol.swift

```diff
@@ -29,6 +30,7 @@ protocol UserSuggestionServiceProtocol {
     var currentTextTrigger: String? { get }
     
     func processTextMessage(_ textMessage: String?)
+    func processSuggestionPattern(_ suggestionPattern: SuggestionPattern?)
```

Comment:
Because we need a way to inform the `UserSuggestionService` that there is no pattern detected anymore (either by selecting user to mention or removing text), so it can "provide" an empty user suggestion list as expected.

In comparison to the legacy `processTextMessage` right above a standard interaction will look something like that :

```swift
processTextMessage("Hello @ali")
[user taps name in list]
processTextMessage("Hello Alice") // => List is removed
```

would be equivalent to 

```swift
processSuggestionPattern(SuggestionPattern([...]))
[user taps name in list]
processSuggestionPattern(nil) // => List is removed
```

Perhaps we could change things a bit to make this more explicit, but this would require changes at the composer library level. Our Rust object around the `SuggestionPattern` look something like that (Swift bindings): 

```swift
public enum MenuAction {
    case keep
    case none
    case suggestion(suggestionPattern: SuggestionPattern)
}
```

The issue being that the other states are information required for library internal operations, It seems simpler for the hosting application to get a Publisher where you either get a SuggestionPattern, or you don't (not sure how worthy it is to re-nest this Struct in a two case enum instead of using Optional).

Reply:
I see, thanks for the explanation. Without making larger refactors this is a pragmatic solution